packages/ts-api-react/src/SessionContext.tsx

```diff
@@ -49,13 +53,29 @@ const USERNAME_KEY = "username";
  */
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function SessionProvider(props: PropsWithChildren) {
+export function SessionProvider(props: Props) {
   const [_session, _setSession] = useState<LoginResponse>();
   const _storage = new StorageManager("Session");
+
+  // Check if the browser has a Django "sessionid" cookie
+  // Same code is repeated inside useValidateSession
```

Comment:
Why? Is this done upon initialization, and `useValidateSession` can be used later if the user logs in?

Reply:
Yep